python/selfie-lib/selfie_lib/LineReader.py

```diff
@@ -0,0 +1,32 @@
+import io
+from typing import Union
+
+class LineReader:
+    def __init__(self, content: Union[bytes, str]):
+        if isinstance(content, str):
+            content = content.encode('utf-8')
+        self.buffer = io.BytesIO(content)
+        self.uses_unix_newlines = self.detect_newline_type()
+
+    def detect_newline_type(self) -> bool:
```

Comment:
This method should be private. See [this chat](https://chat.openai.com/share/bf04c6b6-2750-4985-aeea-dbc8b37ffb63) for details.